agents/pysmurf_controller/pysmurf_controller.py

```diff
@@ -233,33 +257,336 @@ def abort(self, session, params=None):
 
-        **Task** - Runs the fake script /config/scripts/pysmurf/tune_squids.py
+        with self.lock.acquire_timeout(0, job='uxm_setup') as acquired:
+            if not acquired:
+                return False, f"Operation failed: {self.lock.job} is running."
 
-        Args:
-            args (list, optional):
-                List of command line arguments to pass to the script.
-                Defaults to [].
-            log (string/bool, optional):
-                Determines if and how the process's stdout should be logged.
-                You can pass the path to a logfile, True to use the agent's log,
-                or False to not log at all.
+            S, cfg = self._get_smurf_control(session=session)
+            session.set_status('running')
+            self.log.info("Starting UXM setup")
+            success, summary = uxm_setup.uxm_setup(
+                S, cfg, bands=params['bands'], **params['kwargs']
+            )
+            return success, "Finished UXM-setup procedure"
+
+    @ocs_agent.param('bands', default=None)
+    @ocs_agent.param('kwargs', default=None)
+    def uxm_relock(self, session, params):
+        """uxm_relock(bands=None, kwargs=None)
+
+        **Task** - Task to relock detectors to existing tune if setup has
+        already been run. Runs the following operations:
+
+            1. Setup Amps
+            2. Relocks detectors, setup notches (if requested), and serial
+               gradient descent / eta scan
+            3. Tracking setup
+            4. Noise check
+
+        See the `sodetlib relock docs 
+        <https://simons1.princeton.edu/docs/sodetlib/operations/setup.html#relocking>`_
+        for more information on the sodetlib relock procedure and allowed
+        keyword arguments.
+
+        Args
+        -----
+        bands : list, int
+            Bands to set up. Defaults to all.
+        kwargs : dict
+            Dict containing additional keyword args to pass to the uxm_relock
+            function.
+        """
+        if params['kwargs'] is None:
+            params['kwargs'] = {}
+
+        with self.lock.acquire_timeout(0, job='uxm_relock') as acquired:
+            if not acquired:
+                return False, f"Operation failed: {self.lock.job} is running."
 
+            session.set_status('running')
+            S, cfg = self._get_smurf_control(session=session)
+            success, summary = uxm_relock.uxm_relock(
+                S, cfg, bands=params['bands'], **params['kwargs']
+            )
+            return success, "Finished UXM Relock"
+
+    @ocs_agent.param('duration', default=30., type=float)
+    @ocs_agent.param('kwargs', default=None)
+    def take_noise(self, session, params):
+        """take_noise(duration=30., kwargs=None)
+
+        **Task** - Task to take a short timestream and calculate noise
+        statistics. Median white noise level for each band will be stored in
+        the session data. See the `sodetlib noise docs 
```

Comment:
I don't think those noise levels are making it into session.data ...

Reply:
I think it's happening here: https://github.com/simonsobs/sodetlib/blob/86bc88eaa00d7ba0c641077b5ccbe1e72682430b/sodetlib/noise.py#L537

The info stored in `session.data` should be added to this docstring in the `Notes:` section.